native/common/jp_convert.cpp

```diff
@@ -0,0 +1,644 @@
+/*
+ * Copyright 2020 nelson85.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <jpype.h>
+
+static jvalue bool2bool(char* c)
```

Comment:
You could also use an anonymous namespace to avoid declare every function statically. This would look more like c++ than good old c.

Reply:
This was selected because I was arguing with the editor.  The editor has one setting to indent namespaces which it interprets as indenting namepaces and all `extern "C" {` sections.  Thus I either got indents on every sections of C linked code (annoying) and namespaces (which we were using) or not which was creating pointless whitespace diffs.  

Thus in this case I just selected "static" rather than anonymous namespace to avoid the fight.   Personally I would like to see this sort of utility function in CPython rather than re-implementing it here.

Reply:
Fixed.

Reply:
I don't see why these util functions would be desired in CPython. I like the templated version of jp_converter much better!

Reply:
Well the CPython native solution of sending things through a giant switch statement every single time that they want to convert each element is rather poor.  I was hoping to just call their type conversion routine rather than spinning my own.  (Done while watching my sons wrestling meet in the stands.)  But there isn't a lot of portable stuff in CPython in terms of support libraries.  They bury everything as statics even when it is something needed.  That is why you see me write likes like 

```
   PyLong_Type.tp_str(obj)
```

I can't call the actually method directly so I am forced to find someplace in the library where I can guarantee that the virtual table points to something I need to access and call it.  Heaven help us if someone monkey patches it.